server/server.go

```diff
@@ -54,10 +55,133 @@ func (t *TradeServer) buildOrderRoutes(parent *gin.RouterGroup) {
+	var orders []model.PostStockOrderRequest
+
+	if err := c.Bind(&reqBody); err != nil {
+		c.JSON(http.StatusBadRequest, "Bad request.")
+		return
+	}
+	accKeys := reqBody.AccKeys
+	refAccKey := reqBody.RefAccKey
+	if _, ok := accounts[refAccKey]; !ok {
+		c.JSON(http.StatusBadRequest, "Reference account not found.")
+		return
+	}
+	for _, accKey := range accKeys {
+		if accKey == refAccKey {
+			continue
+		}
+		account, ok := accounts[accKey]
+		if !ok {
+			Log(Info, "%s account not found, skipped", accKey)
+			continue
+		}
+		orders = alignPositionRatios(refAccKey, accKey)
+		for _, order := range orders {
+			_, err := account.client.PlaceOrder(account.accountID, order)
+			if err != nil {
+				Log(Error, "Failed to place order for account %s, caught error: %s", accKey, err.Error())
+			}
+		}
+	}
+	c.JSON(http.StatusOK, fmt.Sprintf("Orders placed to align position ratios between %v and %s", accKeys, refAccKey))
+}
+
+func getAccountBalance(accKey string) float32 {
+	balance, err := strconv.ParseFloat(accounts[accKey].accountDetails.NetLiquidation, 32)
+	if err != nil {
+		return -1
+	}
+	return float32(balance)
+}
+
+func getAccountPositions(accKey string) map[int32]int32 {
+	positionsLite := make(map[int32]int32)
+	positions := accounts[accKey].accountDetails.Positions
+	for _, pos := range positions {
+		posCount, _ := strconv.Atoi(pos.Position)
+		positionsLite[pos.Ticker.TickerId] = int32(posCount)
+	}
+	return positionsLite
+}
+
+func getTickerPrice(tickerId int32, accKey string) (float32, error) {
+	quote, err := accounts[accKey].client.GetRealtimeStockQuote(fmt.Sprint(tickerId))
+	if err != nil {
+		return 0, err
+	}
+	price, err := strconv.ParseFloat(quote.PPrice, 32)
+	if err != nil {
+		return 0, err
+	}
+	return float32(price), nil
+}
+
+func alignPositionRatios(refAccKey string, accKey string) []model.PostStockOrderRequest {
+	var orders []model.PostStockOrderRequest
+
+	refPositions := getAccountPositions(refAccKey)
+	refBalance := getAccountBalance(refAccKey)
+	positions := getAccountPositions(accKey)
+	balance := getAccountBalance(accKey)
+
+	allTickers := make(map[int32]struct{})
+	for tickerId := range refPositions {
+		allTickers[tickerId] = struct{}{}
+	}
+	for tickerId := range positions {
+		allTickers[tickerId] = struct{}{}
+	}
+	for tickerId := range allTickers {
+		var refPos int32 = 0
+		var pos int32 = 0
+		if p, ok := refPositions[tickerId]; ok {
+			refPos = p
+		}
+		if p, ok := positions[tickerId]; ok {
+			pos = p
+		}
+
+		tickerPrice, err := getTickerPrice(tickerId, accKey)
+		if err != nil {
+			continue
+		}
+		refPosRatio := tickerPrice * float32(refPos) / refBalance
+		posRatio := tickerPrice * float32(pos) / balance
+		// Only align the positions when the difference is > 1% of the account value
+		if math.Abs(float64(posRatio-refPosRatio)) > 0.01 {
+			targetPos := int32(balance * refPosRatio / tickerPrice)
+			posDelta := targetPos - pos
+			if posDelta != 0 {
```

Comment:
should be if posDelta > 0, then BUY

Reply:
this is already represented in the code below.